analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AnnotateApiActionsWithHttpVerb.cs

```diff
@@ -0,0 +1,86 @@
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    private static readonly ImmutableArray<KnownType> HttpMethodAttributes = ImmutableArray.Create(
+        KnownType.Microsoft_AspNetCore_Mvc_HttpGetAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_HttpPutAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_HttpPostAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_HttpDeleteAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_HttpPatchAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_HttpHeadAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_HttpOptionsAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_Routing_HttpMethodAttribute,
+        KnownType.Microsoft_AspNetCore_Mvc_AcceptVerbsAttribute); // AcceptVerbs is treated as an exception
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(compilationStartContext =>
+        {
+            if (!compilationStartContext.Compilation.ReferencesControllers())
+            {
+                return;
+            }
+
+            compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
+            {
+                var controllerSymbol = (INamedTypeSymbol)symbolStartContext.Symbol;
+                var controllerAttributes = controllerSymbol.GetAttributesWithInherited();
+
+                if (controllerSymbol.IsControllerType()
+                    && controllerAttributes.Any(x => x.AttributeClass.DerivesFrom(KnownType.Microsoft_AspNetCore_Mvc_ApiControllerAttribute))
+                    && !IgnoresApiExplorer(controllerAttributes))
+                {
+                    symbolStartContext.RegisterSyntaxNodeAction(c =>
+                    {
+                        var methodNode = (MethodDeclarationSyntax)c.Node;
+                        var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodNode);
+                        var methodAttributes = methodSymbol.GetAttributesWithInherited();
+
+                        if (methodSymbol.IsControllerMethod()
+                            && !methodSymbol.IsAbstract
+                            && !methodAttributes.Any(x => x.AttributeClass.DerivesFromAny(HttpMethodAttributes))
+                            && !IgnoresApiExplorer(methodAttributes))
+                        {
+                            c.ReportIssue(Rule, methodNode.Identifier.GetLocation());
+                        }
+                    },
+                    SyntaxKind.MethodDeclaration);
+                }
+            },
+            SymbolKind.NamedType);
+        });
+
+    // Tracks [ApiExplorerSettings(IgnoreApi = true)]
+    private static bool IgnoresApiExplorer(IEnumerable<AttributeData> attributes) =>
```

Comment:
There's an `AttributeDataExtensions` class with a method `TryGetAttributeValue` which should do the trick for this.